### ISSUE TITLE
Unidentified model not training reproducibly

### ISSUE THREAD

**Issue by invoxiaehu** [NVIDIA/tensorflow-determinism#3]
or is it?

**Comment by duncanriach** [comment]
tf 1.5 ? Do you mean tf 1.15?

**Comment by invoxiaehu** [comment]
Yes i meant 1.15.0 sorry.
I also had the message confirming the patch.
Then I set the seeds as recommended.
```
os.environ['PYTHONHASHSEED']=str(seed_value)
random.seed(seed_value)
np.random.seed(seed_value)
tf.set_random_seed(seed_value)

```
and train with tf.keras from Sequence.
My first two training weren't reproducible. But if ou say it works for you, I'll check again. 
And if it's confirmed i'll make a small demonstration notebook.
Thank you for your code. Very useful.

**Comment by invoxiaehu** [comment]
Hi,
Unfortunately I couldn't make it work.
I checked with the same CPU code and it's reproducible, so it's not a issue of seed.
Attached the metrics for 20 iterations on both CPU and GPU. We can see that GPU is reproducible for a few steps then diverges. 

Code is not easy to share right now cause uses generators and augmentations, but i'll try to make a basic version.
![image](https://user-images.githubusercontent.com/40634223/68489559-a442db80-0247-11ea-92a0-8353c4461128.jpeg)

**Comment by duncanriach** [comment]
Awesome. Thank you. I'm realizing that there may be some holes in automated integration testing, which I'm addressing. On the other hand, what you're seeing may be because of a configuration issue (e.g. using XLA JIT) or a model issue (e.g. using an op that has not yet been made deterministic).

Thanks for raising this issue, and I'm looking forward to seeing a simplified colab that demonstrates this issue.

**Comment by duncanriach** [comment]
I've run more integration tests and not found an issue. In lieu of you providing a simplified test case, could you confirm that you're not using any ops based on [these kernels](https://github.com/NVIDIA/tensorflow-determinism#other-possible-gpu-specific-sources-of-non-determinism) and also that you're not doing bilinear filtering.

**Comment by duncanriach** [comment]
@invoxiaehu can you provide me with a simple-as-possible, self-contained example that demonstrates the non-determinism, so that it can be reproduced and debugged?

**Comment by duncanriach** [comment]
@invoxiaehu, please will you also check if you are using, either directly or indirectly, `tf.nn.softmax_cross_entropy_with_logits` or `tf.nn.sparse_softmax_cross_entropy_with_logits`? It's looking almost certain that these are injecting non-determinism (as tested in TF 2.0).

**Comment by duncanriach** [comment]
I'm closing this issue. Please feel free to re-open when you can provide code to reproduce the issue so that it can be debugged.
